modules/charon-core/src/main/java/org/wso2/charon3/core/objects/Group.java

```diff
@@ -411,4 +413,104 @@ private void setRole(ComplexAttribute groupPropertiesAttribute) throws CharonExc
         }
     }
 
+    /**
+     * Set the assigned V2 roles of the group.
+     *
+     * @param role RoleV2 object.
+     * @throws CharonException     CharonException.
+     * @throws BadRequestException BadRequestException.
+     */
+    public void setRoleV2(RoleV2 role) throws CharonException, BadRequestException {
+
+        SimpleAttribute valueSimpleAttribute = null;
+        SimpleAttribute displaySimpleAttribute;
+        SimpleAttribute referenceSimpleAttribute;
+        SimpleAttribute audienceValueSimpleAttribute;
+        SimpleAttribute audienceDisplaySimpleAttribute;
+        SimpleAttribute audienceTypeSimpleAttribute;
+        String reference = role.getLocation();
+        String value = role.getId();
+        String display = role.getDisplayName();
+        String audienceValue = role.getAudienceValue();
+        String audienceDisplay = role.getAudienceDisplayName();
+        String audienceType = role.getAudienceType();
+        ComplexAttribute complexAttribute = new ComplexAttribute();
+
+        if (StringUtils.isNotBlank(value)) {
+            valueSimpleAttribute = new SimpleAttribute(SCIMConstants.CommonSchemaConstants.VALUE, value);
+            valueSimpleAttribute = (SimpleAttribute) DefaultAttributeFactory
+                    .createAttribute(SCIMSchemaDefinitions.SCIMGroupSchemaDefinition.ROLES_VALUE, valueSimpleAttribute);
+            complexAttribute.setSubAttribute(valueSimpleAttribute);
+        }
```

Comment:
```suggestion
        if (StringUtils.isNotBlank(value)) {
            complexAttribute.setSubAttribute(getSubAttribute(SCIMConstants.CommonSchemaConstants.VALUE, SCIMSchemaDefinitions.SCIMGroupSchemaDefinition.ROLES_VALUE, value));
        }
        
        private SimpleAttribute getSubAttribute(String a, String b, String c) {
           SimpleAttribute subAttribute =. new SimpleAttribute(a,c);
           return (SimpleAttribute) DefaultAttributeFactory.createAttribute(b, subAttribute);
        }
```

Reply:
fixed in [ca64654](https://github.com/wso2/charon/pull/392/commits/ca64654c3f9c03eb95c521cc2f548d9206cbdb4e)